src/graph/workflow.ts

```diff
@@ -458,11 +458,22 @@ export class Workflow {
     }
 
     private redrawEdges() {
-        Array.from(this.workflow.querySelectorAll(".edge")).forEach(el => {
-            el.remove();
-        });
 
-        const edgesTpl          = this.model.connections.map(c => GraphEdge.makeTemplate(c, this.workflow)).reduce((acc, tpl) => acc + tpl, "");
+        const edgeEls             = this.model.connections.filter(el => el.isVisible);
+        const highlightedEdgesMap = new Map<Edge, boolean>();
+
+        Array.from(this.workflow.querySelectorAll(".edge"))
+            .forEach((el, i) => {
+                const edge = edgeEls[i];
```

Comment:
We can't rely on the order of DOM nodes.
Instead of an edge object, you can store it's connection id (or whatever identifier is available)